internal/models/users/usernames/usernames.go

```diff
@@ -91,3 +96,127 @@ func GetUsernameAdminInfo(db *sql.DB, adjPage int, nounPage int, usernamesPerPag
 
 	return &uai, nil
 }
+
+// AddWordToTable adds a word to the specified table.
+func AddWordToTable(db *sql.DB, word string, tableId string) error {
+	if tableId == "noun-table" {
+		_, err := db.Exec(`INSERT INTO nouns VALUES ($1);`, word)
+		return err
+	} else if tableId == "adjective-table" {
+		_, err := db.Exec(`INSERT INTO adjectives VALUES ($1);`, word)
+		return err
+	} else {
+		//Return error
+		return errors.New("table ID not recognized")
+	}
+}
+
+// DeleteWordsFromTable deletes the words from the adjectives and nouns tables and updates the usernames in the users table.
+func DeleteWordsFromTable(db *sql.DB, ctx context.Context, words []string) error {
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	defer tx.Rollback()
+
+	_, err = tx.ExecContext(ctx, `
+					UPDATE users
+					SET
+						username_adjective = random_username.adjective,
+						username_noun = random_username.noun
+					FROM (
+						SELECT adjective, noun
+						FROM available_usernames 
+						OFFSET floor(random() * (
+						SELECT COUNT(*) FROM available_usernames)
+						) 
+					LIMIT 1) AS random_username
+						WHERE users.id = ( 
+							SELECT id
+							FROM users
+							WHERE username_adjective = ANY($1)
+							OR
+							username_noun = ANY($1)
+						);
+					`, pq.Array(words))
+	if err != nil {
+		return err
+	}
+
+	_, err = tx.ExecContext(ctx, `DELETE FROM adjectives WHERE adjective = ANY($1);`, pq.Array(words))
+	if err != nil {
+		return err
+	}
+	_, err = tx.ExecContext(ctx, `DELETE FROM nouns WHERE noun = ANY($1);`, pq.Array(words))
+	if err != nil {
+		return err
+	}
+
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// UpdateAdjectives updates the adjectives in the adjectives table.
+// The oldNewAdjArr is an 2 dimensional array where each inner array contains two strings: the old adjective and the new adjective.
+//
+// For example:
+//  oldNewAdjArr := [][]string{{"oldAdj1", "newAdj1"}, {"oldAdj2", "newAdj2"}}
+func UpdateAdjectives(db *sql.DB, oldNewAdjArr [][]string) error {
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	defer tx.Rollback()
+
+	pq.Array(oldNewAdjArr)
+
+	for _, oldNewAdj := range oldNewAdjArr {
+		_, err = tx.Exec(`UPDATE adjectives SET adjective = $1 WHERE adjective = $2;`, oldNewAdj[1], oldNewAdj[0])
+		if err != nil {
+			return err
+		}
+	}
+
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+// UpdateNouns updates the nouns in the nouns table.
+// The oldNewNounArr is an 2 dimensional array where each inner array contains two strings: the old noun and the new noun.
+//
+// For example:
+//  oldNewNounArr := [][]string{{"oldNoun1", "newNoun1"}, {"oldNoun2", "newNoun2"}}
```

Comment:
Use a struct instead of a 2D array for readability.

Reply:
done